examples/refreshAccessToken.js

```diff
@@ -0,0 +1,95 @@
+// Copyright 2017, Google, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+const OAuth2Client = require('../build/src/index').OAuth2Client;
+const http = require('http');
+const url = require('url');
+const querystring = require('querystring');
+const opn = require('opn');
+
+// Download your OAuth2 configuration from the Google
+const keys = require('./oauth2.keys.json');
+
+/**
+ * Start by acquiring a pre-authenticated oAuth2 client.
+ */
+async function main() {
+  try {
```

Comment:
nit: use the `main().catch(console.error)` pattern instead?

Reply:
This sent me down a dark rabbit hole from which I've yet to emerge.  I'm having a lot of trouble getting the process to exit naturally, without explicitly calling `process.exit()`.  The http server is hanging onto active connections, so I had to explicitly track all incoming connections, and manually close them to get the server to close.  I'm also running into issues with the `opn` npm module hanging onto it's handle, so I had to do a bunch of hacking to make that work.  In the end, I had added like 50 lines of code to make the process exit naturally, and it really added nothing to the sample.  

Are you ok with me leaving this one the way it is?  I'd be happy to open an issue to dig into this deeper after we get next merged into master.

Reply:
sgtm